webpack/assets/javascripts/react_app/routes/Models/ModelsPage/index.js

```diff
@@ -0,0 +1,67 @@
+import { connect } from 'react-redux';
+import { compose, bindActionCreators } from 'redux';
+import Immutable from 'seamless-immutable';
+
+import ModelsPage from './ModelsPage';
+import * as actions from './ModelsPageActions';
+
+import { callOnMount, callOnPopState } from '../../../common/HOC';
+
+import withDataReducer from '../../common/reducerHOC/withDataReducer';
+
+import {
+  selectModels,
+  selectPage,
+  selectPerPage,
+  selectSearch,
+  selectSort,
+  selectHasData,
+  selectHasError,
+  selectIsLoading,
+  selectSubtotal,
+  selectMessage,
+  selectCanCreate,
+} from './ModelsPageSelectors';
+
+const mapStateToProps = state => ({
+  models: selectModels(state),
+  page: selectPage(state),
+  perPage: selectPerPage(state),
+  search: selectSearch(state),
+  sort: selectSort(state),
+  isLoading: selectIsLoading(state),
+  hasData: selectHasData(state),
+  hasError: selectHasError(state),
+  itemCount: selectSubtotal(state),
+  message: selectMessage(state),
+  canCreate: selectCanCreate(state),
+});
+
+export const initialState = Immutable({
+  total: 0,
```

Comment:
why not use `withQueryReducer` for the pagination and search fields?

Reply:
Because sort order and sort attribute are part of the query when sorting by table column, technically they should be in `withQueryReducer`. That means I would need to extend it with additional state as I do with `withDataReducer`. I would be extending 2 functions in the exact same way to create 2 reducers that handle different part of the same response. But I did not notice any benefits in splitting response payload in 2 different branches of the state tree, I was just dispatching more actions.